### PR TITLE
Update allocations.go

### PR DESCRIPTION
**Pull request by sejalapeno**
updated missing client status "unknown" #17688

Fixes https://github.com/hashicorp/nomad/issues/17688

### CODE REVIEW

**Review by shoenig — APPROVED** [review]

**Comment by shoenig** [comment]
Thanks @sejalapeno! Do you mind amending the PR with a changelog entry? You can create one just by running `make cl`

**Comment by shoenig** [comment]
Ah yeah @sejalapeno it's a little wizzard that prompts a few questions, so that it can generate the properly formatted changelog entry. This is a bug fix, so pick whichever number indicates it's for a bug fix.

**Comment by sejalapeno** [comment]
I hope the changes are what was required

File: .changelog/17726.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+build: Added make target for creating changelog entries
```
Comment (shoenig) [inline .changelog/17726.txt:2]:
```suggestion
api: add missing constant for unknown allocation status
```

@sejalapeno We want a description relevant to the actual change

**Comment by sejalapeno** [comment]
Is there anything from my end that is supposed to be done because the test ui workflow #2 attempt failed

**Comment by shoenig** [comment]
The UI tests are nothing to worry about here

**Pull request merged**